marimo/_server/sessions.py

```diff
@@ -59,13 +59,25 @@
 class QueueManager:
     """Manages queues for a session."""
 
-    def __init__(self, use_multiprocessing: bool):
+    def __init__(self, use_multiprocessing: bool, provide_completions: bool):
         context = mp.get_context("spawn") if use_multiprocessing else None
-        # Control messages for the kernel (run, autocomplete,
-        # set UI element, set config, etc ) are sent through the control queue
-        self.control_queue: QueueType[requests.Request] = (
+
+        # Control messages for the kernel (run, set UI element, set config, etc
+        # ) are sent through the control queue
+        self.control_queue: QueueType[requests.ControlRequest] = (
             context.Queue() if context is not None else queue.Queue()
         )
+
+        # Code completion requests are sent through a separate queue
+        if provide_completions:
+            self.completion_queue: QueueType[
+                requests.CompletionRequest
+            ] | None = (
+                context.Queue() if context is not None else queue.Queue()
+            )
+        else:
+            self.completion_queue = None
```

Comment:
nit: is it expensive to make a queue that never gets used? i wouldnt imagine so. the code would look a lot cleaner / easier to read without `provide_completions`